libass/ass_directwrite.c

```diff
@@ -917,7 +894,10 @@ static void add_font(IDWriteFont *font, IDWriteFontFamily *fontFamily,
 cleanup:
     free(meta.postscript_name);
     free(meta.extended_family);
+    free(font_priv);
```

Comment:
Oh, good catch. I must’ve missed it when this got added. Why `free` this?

Reply:
Oh shoot, I messed up some logic here. Okay so the reasoning is that previously either calloc would fail or it'd be assumed that `ass_font_provider_add_font` call always succeeds, no problem there. But after adding:
```cpp
    HRESULT hr = IDWriteFont_CreateFontFace(font_priv->font, &face);
    if (FAILED(hr) || !face)
        goto cleanup;
```
it now can fail here in between. So if we just went to cleanup like before, it'd leak the memory allocated to font_priv.

What I didn't account for is that the cleanup is of course also ran on successful call and that will cause a problem. Good catch for sure, will correct this